generators/server/templates/gradle/profile_dev.gradle.ejs

```diff
@@ -49,7 +41,7 @@ configurations {
 }
 
 dependencies {
-    developmentOnly "org.springframework.boot:spring-boot-devtools:${springBootVersion}"
+    compileOnly "org.springframework.boot:spring-boot-devtools"
```

Comment:
Out of curiosity, what's the reason for this change?

Reply:
Test with developmentOnly again since its recommended by updated spring-boot.
```suggestion
    developmentOnly "org.springframework.boot:spring-boot-devtools"
```

Reply:
It fails to generate a package with dev profile.
See https://github.com/jhipster/generator-jhipster/actions/runs/8316726401/job/22756588401?pr=25504